users/tests/test_user_login.py

```diff
@@ -22,3 +33,29 @@ def test_user_client_should_not_log_in_when_email_is_none(self):
 
     def test_user_client_should_not_log_in_when_password_is_none(self):
         self.assertFalse(self.client.login(email="testuser@codepoets.it", password=None))
+
+    def test_user_should_redirect_to_change_password_form_if_need_change_password_is_true(self):
+        self._set_need_change_password(need_change_password=True)
+
+        response = self.client.post(reverse("login"), data=self.data)
+        self.assertRedirects(response, reverse("login_password_change"), status_code=302, target_status_code=200)
+
+    def test_user_should_redirect_to_home_if_need_change_password_is_false(self):
+        self._set_need_change_password(need_change_password=False)
+
+        response = self.client.post(reverse("login"), data=self.data)
+        self.assertRedirects(response, reverse("home"), status_code=302, target_status_code=200)
+
+    def test_user_with_correct_email_and_password_should_log_in_on_post(self):
+        response = self.client.post(reverse("login"), data=self.data)
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.user.is_authenticated)
```

Comment:
You should also check to what view this request is redirected. Please add it

Reply:
Isn't it enough to check it in two separate methods, since there are two redirection views depending on need_change_password? Here it is:
```
    def test_user_should_redirect_to_change_password_form_if_need_change_password_is_true(self):
        self._set_need_change_password(need_change_password=True)

        response = self.client.post(reverse("login"), data=self.data)
        self.assertRedirects(response, reverse("login_password_change"), status_code=302, target_status_code=200)

    def test_user_should_redirect_to_home_if_need_change_password_is_false(self):
        self._set_need_change_password(need_change_password=False)

        response = self.client.post(reverse("login"), data=self.data)
        self.assertRedirects(response, reverse("home"), status_code=302, target_status_code=200)
```
But I guess it may me reasonable to check user authentication in both of those tests and discard the one you mentioned.

Reply:
Resolved.